### PR TITLE
Upstream v5.73.0

### PR DESCRIPTION
**Pull request by t0yv0**
Upgrade upstream provider to v5.73.0. 

**New resources:**

imagebuilder/lifecyclePolicy.LifecyclePolicy
resiliencehub/resiliencyPolicy.ResiliencyPolicy
sagemaker/hub.Hub
sagemaker/mlflowTrackingServer.MlflowTrackingServer

**New functions:**

ssm/getPatchBaselines.getPatchBaselines

Fixes: https://github.com/pulumi/pulumi-aws/issues/4665

New patches are taken to compensate for upstream partially removing Go SDKV1, so that Pulumi-only resources such as aws.s3.Bucket continue to work as expected.

### CODE REVIEW

File: patches/0027-Do-not-compute-tags_all-at-TF-level.patch [hunk 1]
```diff
@@ -231,110 +231,20 @@ index 789e374885..d5a771ce81 100644
...
--		}
--	} else if diff.HasChange("tags_all") {
--		if err := diff.SetNewComputed("tags_all"); err != nil {
--			return fmt.Errorf("setting tags_all to computed: %w", err)
--		}
--	}
--
- 	return nil
- }
++	// Pulumi handles tags differently for Plugin Framework.
```
Comment (t0yv0) [inline patches/0027-Do-not-compute-tags_all-at-TF-level.patch:241]: @corymhall yes so as part of this PR I endeavored to minimize subsequent patch conflicts for us. What we want to do is to make this function a no-op. The function keeps changing. If we do this as we did before by deleting the code, the changes generate more conflicts. With this new method we just short-circuit it to nil right upfront, but let the rest of the code drift upstream. I think this is better.

**Comment by t0yv0** [comment]
```
panic: interface conversion: interface {} is nil, not *session.Session
```

Tests pointing out that reverse-engineering AWS SDKv1 removal need to go a bit deeper, working this out.

**Comment by t0yv0** [comment]
ecr/credentials_data_source is maintained by patches and might be affected, need to verify.

**Comment by flostadler** [comment]
@t0yv0 I've added a test for the ECR getCredentials function

File: patches/0044-restore-ECRConn.patch [hunk 2]
```diff
@@ -4,26 +4,26 @@ Date: Wed, 10 Apr 2024 08:49:35 -0400
...
- 	"github.com/hashicorp/terraform-provider-aws/internal/errs"
-@@ -588,6 +589,10 @@ func (c *AWSClient) EC2Client(ctx context.Context) *ec2_sdkv2.Client {
- 	return errs.Must(client[*ec2_sdkv2.Client](ctx, c, names.EC2, make(map[string]any)))
- }
+ 	s3_sdkv1 "github.com/aws/aws-sdk-go/service/s3"
++	"github.com/hashicorp/terraform-provider-aws/internal/errs"
++	"github.com/hashicorp/terraform-provider-aws/names"
+ )
  
+ func (c *AWSClient) S3Conn(ctx context.Context) *s3_sdkv1.S3 {
```
[the remark this answers is not in the document]
Comment (t0yv0) [reply patches/0044-restore-ECRConn.patch:21]: Hmm, indeed, I must have moved some code during rebasing. This does not go well with re-enabling ECR connectivity. I'll fixup.

[the remark this answers is not in the document]
Comment (t0yv0) [reply patches/0044-restore-ECRConn.patch:21]: OK never mind. The patch file is fine. It's hard to read diffs  of patch files. S3Conn is the context, and ECRConn are the added lines. I think this is good to go.

[the remark this answers is not in the document]
Comment (flostadler) [reply patches/0044-restore-ECRConn.patch:21]: Ah yeah you're right. The surrounding context got changed in another patch so the lines show up as new here, but they're not part of the changes of this patch

**Pull request merged**